Marketplace: show two rows in Discover page

raicem opened the pull request:
It depends on the number of items provided in the API response. If it's eight, we'll show the second row.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Add the second row to the Discover page group if the number of provided items are equal to 8. 

Closes https://github.com/Automattic/woocommerce.com/issues/23145.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Connect your test store to local WooCommerce.com instance. Please see PR https://github.com/Automattic/woocommerce.com/pull/23220 for instructions.
2. Checkout [this branch](https://github.com/Automattic/woocommerce.com/tree/raicem/increase-featured-api-endpoint-capacity) from WCCOM. This will allow you to send more items to the in-app discover page.
3. In WCCOM, add more featured products to a **Discover our favorites** group. Add as high as 8. You can try 7, or 6 as well to check if it's breaking the layout.
    - For example, if you make it 7, you'll get a 2x3 items on laptop displays, but on larger displays will show a 1x4. Because we don't have enough items to fill the second row.
5. If you don't have any other groups locally, also add 4 items to **Payments** and **Marketing**. 
6. Give [this URL](https://woocommerce.test/wp-json/wccom-extensions/3.0/featured?no_cache=true) a visit and confirm your new items are appearing there. This updates the cache on the WCCOM side. 
7. Make sure you have the Discover page cache disabled in your test store. I recommend adding this hook:
```php
add_filter('transient_wc_addons_featured', function ($value, $transient) {
	return false;
}, 10, 2);
```
8. Build assets of your local WooCommerce installation. I used `pnpm --filter='@woocommerce/plugin-woocommerce' build`
9. Confirm you can get both two row groups and single row groups in the Discover page. 
10. Change display sizes and confirm you don't get a row that's not fully filled in.
11. Checkout an older tag in the WooCommerce repo, for example 9.6.2 tag, and rebuild assets. This is to confirm bumping up the capacity in the API doesn't break older stores. 
   - so far, I haven't been able to rebuild 9.6.2, because pnpm version errors. Kyle already [explored](https://github.com/Automattic/woocommerce.com/pull/23220#issuecomment-2653579244) this, and confirmed it's working but I'll give it another go. 
<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Show more product cards on Marketplace Discover page
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

kdevnel commented on plugins/woocommerce/client/admin/client/marketplace/components/product-list-content/product-list-content.tsx:
```diff
@@ -28,8 +28,14 @@ export default function ProductListContent( props: {
 	const wccomHelperSettings = getAdminSetting( 'wccomHelper', {} );
 
+	const rowsClassName =
+		props.products.length >= 8
```
I doubt this is worth worrying about, but what if we want to show 3 rows? Is it better to futureproof this to some extent? AI suggestion:

```tsx
const ITEMS_PER_ROW = 4;
const rowCount = Math.ceil( props.products.length / ITEMS_PER_ROW );
const rowsClassName = `woocommerce-marketplace__product-list-content--${
	rowCount === 1 ? 'single-row' : `${ rowCount }-rows`
}`;
```

kdevnel reviewed (commented):
I decided to do an early review even though this is a draft still. Hope that's okay! Overall happy with this approach and it tests okay with my WCCOM branch. Left a code comment (not a dealbreaker) and something else.

I notice on mobile 8 is very long! I wonder if we'd actually want two different experiences on mobile rather than having a long list. Again, maybe something to just consider and feedback for future work.

Not approving just yet until this is ready for a full review.

raicem committed:
Marketplace: show more rows dynamically

We used to show only one row in the Discover and NoResults page. We hid
the extra overflowing elements with CSS.

This changes that logic to use JS instead. So we can show unlimited
amount of rows, without breaking the layout.

kdevnel committed:
Merge branch 'trunk' into update/23145-marketplace-discover-rows

kdevnel commented on plugins/woocommerce/client/admin/client/marketplace/components/product-list-content/product-list-content.tsx:
```diff
@@ -50,18 +46,60 @@ export default function ProductListContent( props: {
 		return () => window.removeEventListener( 'resize', updateColumns );
 	}, [] );
 
+	/**
```
Thank you for these super helpful comments!

kdevnel commented on plugins/woocommerce/client/admin/client/marketplace/components/product-list-content/product-list-content.scss:
Beautiful removal of lots of code ❤️

kdevnel reviewed (approved):
This tests and works perfectly. I personally agree, I think mobile should be limited to 4 per group at all times.

I initially thought this was touching the same code as my group headers PR, but it looks like they don't interact at all! That saves some merging/rebasing.

raicem committed:
Marketplace: limit featured product groups rows on smaller screens

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

raicem commented:
Thank you for the review @kdevnel! I have updated the row count for smaller displays. I was also was able to test on on 9.6.2, using `npx pnpm@9.1.3 install` and `npx pnpm@9.1.3 build`. Confirming it's working as expected.

raja651 committed:
Merge branch 'trunk' into update/23145-marketplace-discover-rows

raicem committed:
Marketplace: remove unused dependency
